src/app/App.tsx

```diff
@@ -8,8 +8,19 @@ import { ArrayParam, NumberParam, useQueryParams, withDefault } from 'use-query-
 import { defaultProductParams } from 'api/product/defaultParams';
 import { GetProductsUrlParams } from 'api/product/product.types';
 import { ProductCategory } from 'api/product/productCategory.enum';
+import { useIntl } from 'react-intl';
+import { validateProductsUrlParams } from 'api/product/product';
 
 export const App = () => {
+  const { formatMessage } = useIntl();
+  const translatedPageTitle = formatMessage({ id: 'page.title' });
+  if (document.title !== translatedPageTitle) {
```

Comment:
To co tutaj robisz to jest side effect. Odpalamy je w `useEffect`, który stoi na straży tego, żeby nie odpalał się ten warunek co rerender komponentu.
A najlepiej zrobić z tego osobnego custom hooka, np `usePageTitle`

Reply:
Słusznie, powinienem na to wpaść sam.